frontend/src/employee-frontend/components/income-statements/IncomeStatementsPage.tsx

```diff
@@ -58,20 +63,67 @@ function IncomeStatementsList({
 export default React.memo(function IncomeStatementsPage() {
   const { i18n } = useTranslation()
 
-  const [result, setResult] = useState<
-    Result<IncomeStatementAwaitingHandler[]>
+  const [areas, setAreas] = useState<Result<CareArea[]>>(Loading.of())
+
+  const [incomeStatements, setIncomeStatements] = useState<
+    Result<Paged<IncomeStatementAwaitingHandler> & { currentPage: number }>
   >(Loading.of())
-  const loadData = useRestApi(getIncomeStatementsAwaitingHandler, setResult)
-  useEffect(() => loadData(), [loadData])
+  const [toggledAreas, setToggledAreas] = useState<string[]>([])
+
+  const loadAreas = useRestApi(getAreas, setAreas)
+  const loadData = useRestApi(
+    getIncomeStatementsAwaitingHandler,
+    setIncomeStatements
+  )
+
+  useEffect(() => {
+    loadAreas()
+    loadData()
+  }, [loadAreas, loadData])
+
+  useEffect(() => {
+    loadData(toggledAreas)
+  }, [loadData, toggledAreas])
```

Comment:
This triggers a duplicate data fetch on page load.

Reply:
👍